tests/Cache/CacheTest.php

```diff
@@ -214,6 +205,38 @@ public function testToJsonThrowsExceptionOnInvalid(): void
 
     private function getSignatureDouble(): SignatureInterface
     {
-        return $this->prophesize(SignatureInterface::class)->reveal();
+        return new class() implements SignatureInterface {
+            public function getPhpVersion(): string
+            {
+                return '7.1.0';
+            }
+
+            public function getFixerVersion(): string
+            {
+                return '2.2.0';
+            }
+
+            public function getIndent(): string
+            {
+                return '    ';
+            }
+
+            public function getLineEnding(): string
+            {
+                return PHP_EOL;
+            }
+
+            public function getRules(): array
+            {
+                return [
+                    "\xB1\x31" => true,
```

Comment:
```suggestion
                    "\xB1\x31" => true, // invalid UTF8 sequence
```

otherwise we lose this detail